python/hail/methods/statgen.py

```diff
@@ -1351,9 +1354,9 @@ def hwe_normalized_pca(dataset, k=10, compute_loadings=False, as_array=False) ->
     info("Running PCA using {} variants.".format(n_variants))
 
     entry_expr = hl.bind(
-        dataset.AC / dataset.n_called,
-        lambda mean_gt: hl.cond(hl.is_defined(dataset.GT),
-                                (dataset.GT.n_alt_alleles() - mean_gt) /
+        dataset.__af,
+        lambda mean_gt: hl.cond(hl.is_defined(entry_expr),
+                                (entry_expr.n_alt_alleles() - mean_gt) /
```

Comment:
this is invalid if you provided an AF expression -- entry_expr has a different source.

Reply:
oh yeah I see. ok, i'll fix once #3211 merges and i can rebase (since there are some changes i'll borrow from there)